app/src/main/java/jp/co/cyberagent/stf/Service.java

```diff
@@ -367,4 +373,73 @@ public void run() {
             }
         }
     }
+
+    /**
+     * Monitors the adb state by checking /sys/class/android_usb/android0/state
+     * <p>
+     * If state is DISCONNECTED then IdentityActivity will be shown to allow easier identification
+     * of malfunctioning devices
+     */
+    private class AdbMonitor extends Thread {
+        // If something goes wrong you have 30 seconds to kill the STFService
+        // Using smaller numbers will basically lock the devices until usb connection is established
+        private static final int INTERVAL_MS = 30000;
+
+        @Override
+        public void run() {
+            Log.d(TAG, "Starting adb monitor thread");
+            java.lang.Process process;
```

Comment:
Btw is there a reason why this can't be declared where it's used? I don't see anything obvious right now unless it's some sort of an optimization.

Reply:
I thought about doing .destroy after everything is finished, but I don't think it's worth it here. Moved to variable init inside the loop